opencti-platform/opencti-graphql/src/database/engine.js

```diff
@@ -2083,35 +2084,116 @@ export const elDeleteElements = async (context, user, elements, stixLoadById) =>
   return dependencyDeletions;
 };
 
-export const prepareElementForIndexing = (element) => {
+const isObject = (value) => {
+  return R.is(Object, value) && Object.keys(value).length > 0;
+};
+
+/**
+ * prepareElementFromDefinition
+ * @param entityType string
+ * @param attribute AttributeDefinition
+ * @param value object
+ * @returns {object}
+ */
+export const prepareElementFromDefinition = (entityType, attribute, value) => {
```

Comment:
True. I think we have to work to prepare the data is expected format and check sooner in the chain. But the target here is to introduce a defensive code to prevent any bad behavior waiting for a better structural approach